namer/consul/src/main/scala/io/buoyant/namer/consul/DcServices.scala

```diff
@@ -62,7 +62,14 @@ private[consul] object DcServices {
             // may consider retrying certain failures (with backoff).
             state() = Activity.Failed(e)
             stats.errors.incr()
-            Future.exception(e)
+            log.debug("consul connection failed: %s", name)
+
+            e match {
```

Comment:
yeah you called it, the failures are flagged as both rejected and nonretryable. Looking around some in finagle I don't see where those are set or how to configure them. RetryFilter/RequeueFilter seem to be the primary users of the nonretryable flag